hw-L11-solid/src/main/java/com/abrenchev/AtmImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.abrenchev;
+
+import com.abrenchev.exceptions.AtmWithdrawException;
+import com.abrenchev.interfaces.Atm;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class AtmImpl implements Atm {
+    private TreeMap<Integer, List<Banknote>> banknoteStorage = new TreeMap<>();
```

Comment:
```suggestion
    private Map<Integer, List<Banknote>> banknoteStorage = new TreeMap<>();
```
Лучше объявлять как интерфейсы. При необходимости проще вносить изменения.

А лучше подумайте надо тем, чтобы ввести отдельный класс для ячейки (Cell), может еще какие-то классы.
```suggestion
    private List<Cell> cells = new List<>();
```

Что будет внутри класса Cell оставляю на ваше усмотрение.

Советую еще раз прочитать задание и на бумажке набросать диаграмму классов UML.
Сразу будет видно, какие классы можно выделить.

Reply:
Done